app/models/grda_warehouse/hud/client.rb

```diff
@@ -1721,6 +1721,14 @@ def date_of_last_homeless_service
       processed_service_history&.last_homeless_date
     end
 
+    def services_for_rollup
+      custom_services.
+        preload(enrollment: [:project, :client], custom_service_type: [:custom_service_category]).
+        preload(:warehouse_project).
```

Comment:
This is probably ok, not sure if preload just added to previous calls or if it merges them, but maybe:
```suggestion
        preload(:warehouse_project, enrollment: [:project, :client], custom_service_type: [:custom_service_category]).
```
just to be safe.

Reply:
As an alternative the the above, and the need to add the relation to the custom service class, it looks like you _could_ just 
```
preload(enrollment: [:client, {project: :warehouse_project], custom_service_type: [:custom_service_category]).
```

I'm not sure that's any better, and you'd have to refer to `enrollment.project.warehouse_project.name(current_user)` to get the appropriate name, but it would be one less cross-over between HMIS & warehouse namespaces.  
I'm happy as it is, just a thought.

Reply:
I committed your suggestion, but would be happy to make this change as well if you think it's preferable to have one less crossover!